testdata/dnn/download_models.py

```diff
@@ -5,215 +5,235 @@
 import os
 import sys
 import tarfile
-import requests
 import shutil
 import argparse
+import time
+from urllib.parse import urlparse
 from pathlib import Path
+try:
+    import requests
+except ImportError:
+    print("This script requires 'requests' library")
+    exit(13)
 
-class Model:
+
+class BuiltinDownloader:
     MB = 1024*1024
     BUFSIZE = 10*MB
+    TIMEOUT = 60
 
-    def __init__(self, **kwargs):
-        self.name = kwargs.pop('name', None)
-        self.url = kwargs.pop('url', None)
-        self.downloader = kwargs.pop('downloader', None)
-        self.filename = Path(kwargs.pop('filename'))
-        self.sha = kwargs.pop('sha', None)
-        self.member = kwargs.pop('member', None)
-        self.sub = kwargs.pop('sub', [])
-        self.gdrive = kwargs.pop('gdrive', None)
+    def print_response(self, response):
+        rcode = response.status_code
+        rsize = int(response.headers.get('content-length', 0)) / self.MB
+        print('    {} [{:.2f} Mb]'.format(rcode, rsize))
 
-    def __str__(self):
-        return 'Model <{}>'.format(self.name)
+    def make_response(self, url, session):
+        pieces = urlparse(url)
+        if pieces.netloc in ["docs.google.com", "drive.google.com"]:
+            return session.get(url, params={'confirm': True}, stream=True, timeout=self.TIMEOUT)
```

Comment:
For some reason this simple thing helps to download large files from Google Drive (Ubuntu 22, packaged requests). I'd appreciate additional checks performed by other people using different OSes and locations.